contracts/utils/VaultQueue.sol

```diff
@@ -0,0 +1,228 @@
+// SPDX-License-Identifier: MIT
+pragma solidity =0.8.4;
+
+import {
+    Initializable
+} from "@openzeppelin/contracts-upgradeable/proxy/utils/Initializable.sol";
+import {
+    OwnableUpgradeable
+} from "@openzeppelin/contracts-upgradeable/access/OwnableUpgradeable.sol";
+import {
+    ReentrancyGuardUpgradeable
+} from "@openzeppelin/contracts-upgradeable/security/ReentrancyGuardUpgradeable.sol";
+import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import {
+    SafeERC20
+} from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import {ISAVAX} from "../interfaces/ISAVAX.sol";
+import {IRibbonVault, IDepositContract} from "../interfaces/IRibbon.sol";
+import {ICRV} from "../interfaces/ICRV.sol";
+import {Vault} from "../libraries/Vault.sol";
+import {DSMath} from "../vendor/DSMath.sol";
+
+contract VaultQueue is
+    Initializable,
+    OwnableUpgradeable,
+    ReentrancyGuardUpgradeable
+{
+    enum TransferAction {INTERVAULT, WITHDRAW}
+    struct Transfer {
+        address creditor;
+        address srcVault;
+        address dstVault;
+        address depositContract;
```

Comment:
Maybe we can completely remove the need to store the `depositContract` by hardcoding the mapping of `dstVault` to the DepositHelper contract. Since this edge case only exists for savax currently

Reply:
I think the upside of leaving it here to support future contracts outweighs the minimal gas we'll spend removing it

Reply:
Removed dstVault